collector.go

```diff
@@ -88,10 +89,19 @@ func (c *ErrorCollector) getAggregatedErrors() payload {
 	return payload{aggregatedErrors}
 }
 
-func (c *ErrorCollector) addError(err error, httpCtx *HTTPContext) {
+// getAggregationKey gets the aggregation key of the error
+func getAggregationKey(errorWithContext errorWithContext, errKey ...string) string {
+	if len(errKey) > 0 {
+		// aggregate also by error type
+		return fmt.Sprintf("%s@%s", errorWithContext.Error.Class, errKey[0])
```

Comment:
If we always return the 1st error key why do we expect variadic arguments?

Reply:
I used variadic arguments because there isn't any way in Go to have default parameters. The other option is duplicate all the `Report` functions.

Reply:
I see. Thanks for the explanation :)

Reply:
Hey @marctc. I had to go and dig through PRs to find this explanation, because I was wondering the same thing. 😄 
As a package user, I found it confusing that the function supports a variadic parameter, but only uses the first key. I'm happy I found that out before actually using it to report errors using error keys.

Looking back at this decision now, do you think this is still the right call? I think that package users would benefit from a clearer API, without having to dig through the code to see how the parameters are actually used.

The way I see it, we can clean this up in 2 different ways:

1. As already mentioned, duplicate all the `Report` functions so that we have one version with `errKey string` and one without.
2. Introduce a new struct `ErrorReport` with something like:
    ```go
    type ErrorReport struct {
        err error
        severity Severity
        httpCtx *HTTPContext
        errKey string
    }
    ```
    And use that to introduce `func (c *ErrorCollector) Report(errorReport ErrorReport)`.

What do you think?

Reply:
Hey @alonpeer, thanks for your feedback. I'm not sure if it was the right call, but I'm my pythonic mind it made total sense :stuck_out_tongue:. I would prefer to go to the first way as the second one would break existing changes and it would make the usage of `Report` a bit more awkward. Wdyt?

Reply:
@marctc Sounds good to me. I'll work on a PR for this.

Reply:
Actually, I'm not sure what the best way to accomplish it would be. Go doesn't support function overloading, so I can't do this:

```go
func (c *ErrorCollector) Report(err error) { ... }
func (c *ErrorCollector) Report(err error, errKey string) { ... }
```

So if we want to create multiple functions, we'll have to name them differently, which will also break the current API, since some clients would have to switch to use the other functions (e.g. `ReportWithErrKey()` instead of `Report()`).

I think we can go with option 2, but then also keep the other functions around to allow "non-awkward" use of the API. So we can have something like this:

```go
func (c *ErrorCollector) Report(errorReport ErrorReport)
func (c *ErrorCollector) ReportError(err error)
func (c *ErrorCollector) ReportWithSeverity(err error, severity Severity)
func (c *ErrorCollector) ReportWithHTTPContext(err error, httpCtx *HTTPContext)
// ...
```

Since we're going to break the API anyway, might as well do it in a way that is future-proof. We might end up adding more data to the report, but we can keep the interface the same with the `ErrorReport` struct.

Reply:
https://github.com/periskop-dev/periskop-go/pull/12